### PR TITLE
Add Package Manager Basic Support.

### PR DESCRIPTION
**Pull request by orangebees**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [x] N
- [ ] Y 

#### 2. What is the scope of this PR (e.g. component or file name):
Add Package Manager Basic Support.

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [x] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Add Package Manager Basic Support.Initial commands support for init, add, and del. You can add packages from GitHub

# Depends on import conventions
Kpm conventions add dependencies via kpm via import external. . <sub_pkg>Import dependencies

Kpm defaults to the current project working directory externally as a dependent placement directory, which is equivalent to the node_modules of nodejs, and the required git packages are softlinked from kpm /git/kcl_modules/<pkgname@version#commitid> to the current project working directory external.

# Add dependencies
If the package is added through git, first put the package git clone to the cache directory, then scan the file to generate a description metadata of the package, pkginfo.json, add the package file in the global store, and then use pkginfo.json to kcl_modules in kpm /git/<pkgname@version#commitid> The directory generates a project package obtained by restoring the hard link in the kpm global store, and finally writes pkginfo.json to kpm/git/metadata/<pkgname@version#commitid>.json.

```release-note
Example:
/app # kcl-go kpm init testproject
Create kpm.json success!
/app  # ls
kcl.mod   kpm.json
/app # kcl-go  kpm add -git https://github.com/KusionStack/konfig
not found pkg https://github.com/KusionStack/konfig@v0.0.0#94719ae17ed16b27f4b21ab2d6560dd0023cadd3
building https://github.com/KusionStack/konfig@v0.0.0#94719ae17ed16b27f4b21ab2d6560dd0023cadd3
/app # ls external
konfig
/app # kcl-go  kpm del konfig
del  dependencies konfig  success
```

### CODE REVIEW

**Comment by orangebees** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Peefy** [comment]
Need Go 1.18 set cc @chai2010

File: pkg/kpm/Set.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+func (s Set) SRem(str string) error {
+	_, err := s[str]
+	if err {
+
+	}
+	delete(s, str)
+	return nil
+}
+
+// SUnion 并集
```
Comment (chai2010) [inline pkg/kpm/Set.go]: remove Chinese comment

File: pkg/kpm/Set.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+func (s Set) Reset() {
+	for s2, _ := range s {
+		delete(s, s2)
+	}
+}
+
+var pool = sync.Pool{
+	New: func() any {
```
Comment (chai2010) [inline pkg/kpm/Set.go]: s/any/interface{}

File: cmds/kcl-go/command/cmd_kpm.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package command
```
Comment (chai2010) [inline cmds/kcl-go/command/cmd_kpm.go:1]: all: add copyright comment

File: pkg/kpm/var.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package kpm
+
+// KPM_ROOT kpm工作目录
+var KPM_ROOT = "C:\\kpmroot"
+
+// 当前工作目录
+var pwd = ""
+
+//var err error = nil
```
Comment (chai2010) [inline pkg/kpm/var.go]: remove unused code

File: pkg/kpm/version.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+	"strings"
+)
+
+type Version struct {
+	Major int
+	Minor int
+	Patch int
+}
+
+func (v Version) ToString() string {
```
Comment (chai2010) [inline pkg/kpm/version.go]: use `fmt.Stringer` interface, s/ToString/String

**Comment by Peefy** [comment]
Pls open PRs at https://github.com/KusionStack/kpm. cc @orangebees

**Pull request closed without merging**